packages/app/src/runner/event-manager.ts

```diff
@@ -673,6 +671,28 @@ export class EventManager {
       // this will trigger a log changed event for the log itself.
       log?.set(attrs)
     })
+
+    // The window.top should not change between test reloads, and we only need to bind the message event when Cypress is recreated
+    // Forward all message events to the current instance of the multi-origin communicator
+    if (!window.top) throw new Error('missing window.top in event-manager')
+
+    /**
+     * NOTE: Be sure to remove the cross origin onMessage bus to make sure the communicator doesn't live on inside a closure and cause tied up events.
+     *
+     * This is applicable when a user navigates away from the runner and into the "specs" menu or otherwise,
+     * and the EventManager is recreated. This is the main reason this reference is scoped to the file and NOT the instance.
+     *
+     * This is also applicable when a user changes their spec file and hot reloads their spec, in which case we need to rebind onMessage
+     * with the newly creates Cypress.primaryOriginCommunicator
+     */
+    window?.top?.removeEventListener('message', crossOriginOnMessageRef, false)
+    crossOriginOnMessageRef = ({ data, source }) => {
+      Cypress?.primaryOriginCommunicator.onMessage({ data, source })
+
+      return undefined
+    }
```

Comment:
Could we just move this entire thing to the outer scope and rely on the `Cypress` being global, rather than reassigning?

Reply:
We can't due to `Cypress` being recreated. What ultimately happens is the `primaryOriginCommunicator` is the wrong instance, so we need to rebind when Cypress is created